satpy/multiscene/_multiscene.py

```diff
@@ -359,6 +359,10 @@ def blend(
             datasets = [scn[ds_id] for scn in self.scenes if ds_id in scn]
             new_scn[ds_id] = blend_function(datasets)
 
+        new_scn._wishlist = self.first_scene.wishlist.copy()
+        # without copying this, there is a KeyError in dependency_tree.trunk
+        new_scn._dependency_tree = self.first_scene._dependency_tree.copy()
+        new_scn.generate_possible_composites(True)
```

Comment:
Yeah maybe we do basically need to copy the first Scene just like `Scene.copy` does:

https://github.com/pytroll/satpy/blob/3fd4c458cab81df54043b16de7e43640a6b6e72f/satpy/scene.py#L562-L576

Reply:
The failure I get if not copying the dependency tree:

```
../../multiscene/_multiscene.py:365: in blend
    new_scn.generate_possible_composites(True)
../../scene.py:1430: in generate_possible_composites
    keepables = self._generate_composites_from_loaded_datasets()
../../scene.py:1446: in _generate_composites_from_loaded_datasets
    trunk_nodes = self._dependency_tree.trunk(limit_nodes_to=self.missing_datasets,
../../dependency_tree.py:97: in trunk
    child_node = self._all_nodes[child_id]
../../dependency_tree.py:615: in __getitem__
    key = self.get_key(item)
../../dependency_tree.py:607: in get_key
    return get_key(match_key, self.keys())
_



        key = create_filtered_query(key, query)

        res = key.filter_dataids(key_container)
        if not res:
>           raise KeyError("No dataset matching '{}' found".format(str(key)))
E           KeyError: "No dataset matching 'DataQuery(name='temporal')' found"

../../dataset/data_dict.py:107: KeyError
```